xbmc/interfaces/legacy/AddonUtils.h

```diff
@@ -68,6 +69,9 @@ namespace XBMCAddonUtils
   };
 
 #define LOCKGUI XBMCAddonUtils::GuiLock __gl
+#define LOCKGUIIF(cond) std::unique_ptr<XBMCAddonUtils::GuiLock> __gl; \
```

Comment:
LOCKGUIIF(cond)
vs
XBMCAddonUtils::GuiLock _gl(cond)

I really don't see an advantage of a macro here. 

But this is not the actual problem. The ctor of GuiLock also exits the GIL. If you passed cond as an argument to the ctor, you could leave the gui lock but still exit gil.